src/main/java/kitchenpos/menus/tobe/domain/Menu.java

```diff
@@ -0,0 +1,131 @@
+package kitchenpos.menus.tobe.domain;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.UUID;
+
+@Table(name = "menu")
+@Entity(name = "TobeMenu")
+public class Menu {
+    @Column(name = "id", columnDefinition = "varbinary(16)")
+    @Id
+    private UUID id;
+
+    @Embedded
+    private MenuName menuName;
+
+    @Embedded
+    private MenuPrice menuPrice;
+
+    @ManyToOne(optional = false)
+    @JoinColumn(
+            name = "menu_group_id",
+            columnDefinition = "varbinary(16)",
+            foreignKey = @ForeignKey(name = "fk_menu_to_menu_group")
+    )
+    private MenuGroup menuGroup;
+
+    @Column(name = "displayed", nullable = false)
+    private boolean displayed;
+
+    @Embedded
+    private MenuProducts menuProducts;
+
+    @Transient
+    private UUID menuGroupId;
+
+    protected Menu() {}
+
+    private Menu(final UUID id, final MenuName menuName, final MenuPrice menuPrice, final MenuGroup menuGroup, final boolean displayed, final MenuProducts menuProducts, final UUID menuGroupId) {
+        this.id = id;
+        this.menuName = menuName;
+        this.menuPrice = menuPrice;
+        this.menuGroup = menuGroup;
+        this.displayed = displayed;
+        this.menuProducts = menuProducts;
+        this.menuGroupId = menuGroupId;
+    }
+
+    public Menu(final MenuName menuName, final MenuPrice menuPrice, final MenuGroup menuGroup, final boolean displayed, final MenuProducts menuProducts, final UUID menuGroupId) {
+        validateMenu(menuName, menuPrice, menuGroup, displayed, menuProducts, menuGroupId);
+        this.id = UUID.randomUUID();
+        this.menuName = menuName;
+        this.menuPrice = menuPrice;
+        this.menuGroup = menuGroup;
+        this.displayed = displayed;
+        this.menuProducts = menuProducts;
+        this.menuGroupId = menuGroupId;
+    }
+
+    public UUID getId() {
+        return id;
+    }
+
+    public String getName() {
+        return menuName.getName();
+    }
+
+    public BigDecimal getPrice() {
+        return menuPrice.getPrice();
+    }
+
+    public MenuGroup getMenuGroup() {
+        return menuGroup;
+    }
+
+    public boolean isDisplayed() {
+        return displayed;
+    }
+
+    public List<MenuProduct> getMenuProducts() {
+        return menuProducts.getMenuProducts();
+    }
```

Comment:
menuProducts 의 주솟값을 그대로 받아 사용하면 어떤 문제가 발생할까요?

Reply:
```suggestion
    public MenuProducts getMenuProducts() {
        return menuProducts;
    }
```

전달할 때 `List<MenuProduct>` 가 아니라 일급 컬렉션 그 자체인 `MenuProducts` 를 리턴하는 편이 더 좋을 까요?

> menuProducts 의 주솟값을 그대로 받아 사용하면 어떤 문제가 발생할까요?

제가 일부러 `return menuProducts` 를 하지 않고, `return menuProducts.getMenuProducts();` 를 하는 이유는,
넘겨받는 입장에서, 타고들어가는 체이닝이 너무 많이 발생해서, 디미터 법칙 관점에서 안 좋아보이길래,
지금과 같이 했습니다.

사실 MenuProducts 라는 일급 컬렉션을 만들지 말고, Domain Service 를 만들었으면 이런 문제가 생기지 않았을 것 같네요.

Reply:
제가 말씀드린 부분은 방어적 복사에 해당하는데요~

https://velog.io/@rito/%EB%B0%A9%EC%96%B4%EC%A0%81-%EB%B3%B5%EC%82%AC-vs-Unmodifiable-Collection

아래의 글을 참고하시면 좋을 것 같아요